flink-contrib/flink-storm/src/test/java/org/apache/flink/storm/api/FlinkTopologyTest.java

```diff
@@ -14,50 +14,70 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package org.apache.flink.storm.api;
 
-import org.apache.flink.storm.api.FlinkTopology;
-import org.junit.Assert;
+
+import backtype.storm.topology.TopologyBuilder;
+import backtype.storm.tuple.Fields;
+import org.apache.flink.storm.util.TestDummyBolt;
+import org.apache.flink.storm.util.TestDummySpout;
+import org.apache.flink.storm.util.TestSink;
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class FlinkTopologyTest {
 
-	@Test
-	public void testDefaultParallelism() {
-		final FlinkTopology topology = new FlinkTopology();
-		Assert.assertEquals(1, topology.getParallelism());
+	@Test(expected = RuntimeException.class)
+	public void testUnknowSpout() {
+		TopologyBuilder builder = new TopologyBuilder();
+		builder.setSpout("spout", new TestSpout());
+		builder.setBolt("bolt", new TestBolt()).shuffleGrouping("unknown");
+
+		FlinkTopology.createTopology(builder);
 	}
 
-	@Test(expected = UnsupportedOperationException.class)
-	public void testExecute() throws Exception {
-		new FlinkTopology().execute();
+	@Test(expected = RuntimeException.class)
+	public void testUnknowBolt() {
+		TopologyBuilder builder = new TopologyBuilder();
+		builder.setSpout("spout", new TestSpout());
+		builder.setBolt("bolt1", new TestBolt()).shuffleGrouping("spout");
+		builder.setBolt("bolt2", new TestBolt()).shuffleGrouping("unknown");
+
+		FlinkTopology.createTopology(builder);
 	}
 
-	@Test(expected = UnsupportedOperationException.class)
-	public void testExecuteWithName() throws Exception {
-		new FlinkTopology().execute(null);
+	@Test(expected = RuntimeException.class)
+	public void testUndeclaredStream() {
+		TopologyBuilder builder = new TopologyBuilder();
+		builder.setSpout("spout", new TestSpout());
+		builder.setBolt("bolt", new TestBolt()).shuffleGrouping("spout");
+
+		FlinkTopology.createTopology(builder);
 	}
 
 	@Test
-	public void testNumberOfTasks() {
-		final FlinkTopology topology = new FlinkTopology();
+	@Ignore
+	public void testFieldsGroupingOnMultipleSpoutOutputStreams() {
+		TopologyBuilder builder = new TopologyBuilder();
 
-		Assert.assertEquals(0, topology.getNumberOfTasks());
+		builder.setSpout("spout", new TestDummySpout());
+		builder.setBolt("sink", new TestSink()).fieldsGrouping("spout",
+				TestDummySpout.spoutStreamId, new Fields("id"));
 
-		topology.increaseNumberOfTasks(3);
-		Assert.assertEquals(3, topology.getNumberOfTasks());
+		FlinkTopology.createTopology(builder);
+	}
 
-		topology.increaseNumberOfTasks(2);
-		Assert.assertEquals(5, topology.getNumberOfTasks());
+	@Test
+	@Ignore
```

Comment:
Please enable this test. I forgot to do this in my last commit which fixes this issue...